### PR TITLE
Remote image plugin

### PR DESCRIPTION
**Pull request by jbukuts**
Added a remark Gatsby plugin to transform relative image paths to specify absolute remote URLs. This works by altering the Markdown AST in place before the nodes for each file are inserted into Gatsby's GQL layer.  This is slightly different than a normal remark plugin given we need the context of the Markdown file's path in the project to build the full link. 

Content changes:

- N/A

Code changes:

- Added a [gatsby-remark-plugin](https://www.gatsbyjs.com/tutorial/remark-plugin-tutorial/) to convert our relative image links to refer to the actually hosted images on GitHub
   - This should reduce deployment time given there would no longer be a need to copy over all images into the `gh-pages` branch
   - This can also be converted and used if we separate our content from the site code.  
- Also removed some old code that's no longer needed 

Testing:
- If you run `npm start` it should default to build a static site using remote images
   - Test to ensure no images a missing. Inspecting an image element will show its `src` as a remote URL 
   - Feel free to also test silly image paths that reference parent directories multiple times like `../test/images/../images/101.png`
- Running `npm run clean && npm run build:local && npm run serve` is the functional equivalent of having the images included in the static build
- Test both of these and compare the `public` folders size to see the difference

### CODE REVIEW

File: package-lock.json
Comment (jbukuts) [inline package-lock.json:1]: Only added `cross-env`. Not sure why so large

File: src/plugins/gatsby-remark-remote-git-images/index.js
Comment (jbukuts) [inline src/plugins/gatsby-remark-remote-git-images/index.js:1]: This is the actual logic of the plugin. Essentially it parses over each markdown file's AST and searches for image tags that have relative URLs. 

If one is found that URL is overwritten with an absolute remote URL.

File: src/plugins/gatsby-remark-remote-git-images/package.json
Comment (jbukuts) [inline src/plugins/gatsby-remark-remote-git-images/package.json:1]: In order to use this within the gatsby-config it needs to be seen as it's own package.

File: gatsby-config.mjs [hunk 1]
```diff
@@ -6,15 +6,34 @@ import rehypeHighlight from 'rehype-highlight';
...
+
+const remoteImagesSettings = {
+  resolve: require.resolve(`./src/plugins/gatsby-remark-remote-git-images`),
+  options: {
+    user: 'ibm-build-lab',
+    repo: 'VAD-VAR-Workshop',
+    branch: 'main',
+    pathPrefix: 'content'
+  }
+};
```
Comment (jbukuts) [inline gatsby-config.mjs:25]: This is how the plugin is consumed.

File: src/plugins/gatsby-remark-remote-git-images/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    (node) => {
+      const { url: imagePath } = node;
+
+      // build full path and march through to handle paths referencing parent directory
+      node.url = [BASE_URL]
+        .concat(
+          cleanPathString(imagePath)
+            .split('/')
+            .reduce(
+              (acc, curr) => (curr !== '..' ? acc.concat(curr) : acc.slice(0, -1)),
```
Comment (jbukuts) [inline src/plugins/gatsby-remark-remote-git-images/index.js:39]: This logic handles the possibility of a relative path referencing a parent directory like `../test/images/101.png`

File: package-lock.json
Comment (ridgill) [reply package-lock.json:1]: wonder if someone had previously updated after building with a different node/npm version?

File: src/helpers/helpers.mjs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    .join('/');
+};
+
+/**
+ * Determines if a URL string is remote
+ * @param {string} url - URL string to determine if it's remote/absolute
+ * @returns {boolean}
+ */
+const isAbsoluteUrl = (url) => {
+  // const ABSOLUTE_URL_REGEX = /^[a-zA-Z][a-zA-Z\d+\-.]*?:/;
```
Comment (ridgill) [inline src/helpers/helpers.mjs:19]: Can remove?

File: src/helpers/helpers.mjs
Comment (jbukuts) [inline src/helpers/helpers.mjs:1]: The reason for the `.mjs` change is that `gatsby-config` does not yet, to my knowledge, support Typescript. This means the remark plugin created as well as the helper function it uses can't be in TS.

**Comment by ridgill** [comment]
I pulled down and checked all images over the weekend, all showing up as expected, referencing the remote github url.  Looks great to me!

File: package-lock.json
Comment (jbukuts) [reply package-lock.json:1]: Tried pulling the file from main and reinstalling `cross-env`. Still resulted in the large change so I guess it was needed.

**Review by ridgill — APPROVED** [review]
LGTM, nice update!

**Pull request merged**